alphagov/paas-cf
# [#127449817] Convert tag_release.sh test to Ginkgo

[#127449817 Convert tests in concourse/scripts/ to Ginkgo + Gomega](https://www.pivotaltracker.com/story/show/127449817)

What
====

We want to convert all the tests to Ginkgo so that we can remove Aruba as described in 7f79755.

The new script uses ginkgo gexec library, and a bunch of helper methods has
been implemented to execute external commands. The most important is
`bashScript` and `runBashScript`, which will execute bash with the flags
`-e`, fail on error and `-u` fail on unassigned variable, and will check if
the command finished successfully.

This test, as the original one in rspec,  relies on executing external
scripts to do git operations on the repos, and check the resulting status.

The test will create a temporary directory.
 * `$HOME` will be set to such directory, as it is required to change the
global config of git for the tag_release.sh.
 * All required files must be copied there, including the script to test.
 * All the scripts will run in that directory.

How to review
------------

 * All the tests in travis should pass.
 * install golang 1.6 and [ginkgo](https://onsi.github.io/ginkgo/) and run
   the tests: `cd concourse/scripts; go test -ginkgo.v tag_release_test.go`
 * Check the code


Who?
----

Anyone but @keymon
